src/containers/MarketsTable/index.tsx

```diff
@@ -1,182 +1,104 @@
-                redirectToTrading={this.handleRedirectToTrading}
-                setCurrentBidUnit={this.handleSetCurrentBidUnit}
-                translate={this.translate}
-            />
-        );
+    if (markets.length > 0) {
+        currentBidUnitsList = markets.reduce(formatFilteredMarkets, currentBidUnitsList);
     }
 
-    private handleRedirectToTrading = (id: string) => {
-        const { markets } = this.props;
-        const currentMarket: Market | undefined = markets.find(item => item.id === id);
+    let currentBidUnitMarkets = markets;
 
-        if (currentMarket) {
-            this.props.setCurrentMarket(currentMarket);
-            this.props.history.push(`/trading/${currentMarket.id}`);
-        }
-    };
-
-    private handleSetCurrentBidUnit = (currentBidUnit?: string) => {
-        this.setState({
-            currentBidUnit: currentBidUnit || '',
-        });
-    };
-
-    private formatFilteredMarkets = (list: string[], market: Market) => {
-        if (list.indexOf(market.quote_unit) === -1) {
-            list.push(market.quote_unit);
-        }
-
-        return list;
-    };
-
-    private translate = (key: string) => this.props.intl.formatMessage({id: key});
-}
-
-const mapStateToProps = (state: RootState): ReduxProps => ({
-    loggedIn: selectUserLoggedIn(state),
-    markets: selectMarkets(state),
-    marketTickers: selectMarketTickers(state),
-    rangerState: selectRanger(state),
-});
-
-const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = dispatch => ({
-    fetchMarkets: () => dispatch(marketsFetch()),
-    rangerConnect: (payload: RangerConnectFetch['payload']) => dispatch(rangerConnectFetch(payload)),
-    setCurrentMarket: (market: Market) => dispatch(setCurrentMarket(market)),
-    tickers: () => dispatch(marketsTickersFetch()),
-});
+    if (currentBidUnit) {
+        currentBidUnitMarkets = markets.length ? markets.filter(market => market.quote_unit === currentBidUnit) : [];
+    }
 
-// tslint:disable-next-line:no-any
-export const MarketsTable = injectIntl(withRouter(connect(mapStateToProps, mapDispatchToProps)(MarketsTableContainer) as any)) as any;
+    const formattedMarkets = currentBidUnitMarkets.length ? currentBidUnitMarkets.map(market =>
+        ({
+            ...market,
+            last: Decimal.format(Number((marketTickers[market.id] || defaultTicker).last), market.amount_precision),
+            open: Decimal.format(Number((marketTickers[market.id] || defaultTicker).open), market.price_precision),
+            price_change_percent: String((marketTickers[market.id] || defaultTicker).price_change_percent),
+            high: Decimal.format(Number((marketTickers[market.id] || defaultTicker).high), market.amount_precision),
+            low: Decimal.format(Number((marketTickers[market.id] || defaultTicker).low), market.amount_precision),
+            volume: Decimal.format(Number((marketTickers[market.id] || defaultTicker).volume), market.amount_precision),
```

Comment:
it can be saved to a variable
`const ticker = marketTickers[market.id] || defaultTicker`
Hence less code


I know it was here before 🙄

Reply:
will be refactored later too

Reply:
will be refactored in future.
Need to create global object that will contain merge of Market & Ticker